cli/proc_state.rs

```diff
@@ -343,6 +343,30 @@ impl ProcState {
       None,
     )
     .await;
+
+    let needs_cjs_esm_translation = graph
+      .modules()
+      .iter()
+      .any(|m| m.kind == ModuleKind::CommonJs);
+
+    if needs_cjs_esm_translation {
+      for module in graph.modules() {
+        if module.kind == ModuleKind::CommonJs {
```

Comment:
@kitsonk I just hit a very strange problem here. Before running tests I did `export DENO_NODE_COMPAT_URL=file:///Users/ib/dev/deno_std/` to use local copy of `deno_std` instead of pulling sources from `deno.land/std/` and somehow I get following files marked as `CommonJs`:
```
CJS module file:///Users/ib/dev/deno_std/node/_buffer.js
CJS module file:///Users/ib/dev/deno_std/node/_events.js
CJS module file:///Users/ib/dev/deno_std/node/_process/stdio.js
CJS module file:///Users/ib/dev/deno_std/node/_process/streams.js
CJS module file:///Users/ib/dev/deno_std/node/_stream.js
CJS module file:///Users/ib/dev/deno_std/node/internal/blob.js
CJS module file:///Users/ib/dev/deno_std/node/internal/buffer.js
CJS module file:///Users/ib/dev/deno_std/node/internal/normalize_encoding.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/add-abort-signal.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/buffer_list.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/compose.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/destroy.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/duplex.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/end-of-stream.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/from.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/legacy.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/passthrough.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/pipeline.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/readable.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/state.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/transform.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/utils.js
CJS module file:///Users/ib/dev/deno_std/node/internal/streams/writable.js
CJS module file:///Users/ib/dev/deno_std/node/internal/timers.js
CJS module file:///Users/ib/dev/deno_std/node/internal/util.js
CJS module file:///Users/ib/dev/deno_std/node/internal/util/inspect.js
CJS module file:///Users/ib/dev/deno_std/node/internal/validators.js
CJS module file:///Users/ib/dev/deno_std/node/stream/promises.js
CJS module file:///Users/ib/dev/deno_std/node/util/types.js
```
While in fact all of these files are ES modules. This doesn't happen if sources are pulled from `deno.land/std/`. Any idea what might be going wrong here?

Reply:
The only way they get marked as [`CommonJS` is via the resolver I believe](https://github.com/denoland/deno_graph/blob/20ae69f4eb3646e48845ec09922e2df792261d01/src/graph.rs#L357-L362), so I would look at logging out calls to the resolver and see why the resolver is returning a CommonJS resolve result.

Reply:
Ah of course 🤦 in this case they are marked as `CommonJs` because they are local files and have `.js` extension (instead of `.mjs` or relevant `type` entry in `package.json`). This is going to be a bit problematic in the long run, I feel like we might be "safer" to change these files in `deno_std` to use `.mjs` extension.